include/support.h

```diff
@@ -185,6 +186,30 @@ void upcase(std::string &str);
 void lowcase(std::string &str);
 void strip_punctuation(std::string &str);
 
+// Split a string on an arbitrary character delimiter. Absent string content on
+// either side of a delimiter is treated as an empty string. For example:
+//   split("abc:", ':') returns {"abc", ""}
+//   split(":def", ':') returns {"", "def"}
+//   split(":", ':') returns {"", ""}
+//   split("::", ':') returns {"", "", ""}
+std::vector<std::string> split(std::string const &seq, const char delim);
```

Comment:
```C++
std::vector<std::string> split(const std::string &seq, char delim);
```
Put const before type for reference and no const for parameters passed by-value in function declarations. Function definition can stay as it is - it's good :)

Reply:
Argh - I caught that yesterday, but maybe only updated in the .cpp and didn't get the header. Will do!